packages/connect-conformance/src/web/transport.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { createRegistry } from "@bufbuild/protobuf";
+import { ClientCompatRequest } from "../gen/connectrpc/conformance/v1/client_compat_pb.js";
+import {
+  Codec,
+  HTTPVersion,
+  Protocol,
+  Compression as ConformanceCompression,
+} from "../gen/connectrpc/conformance/v1/config_pb.js";
+import {
+  createConnectTransport,
+  createGrpcWebTransport,
+} from "@connectrpc/connect-web";
+import {
+  BidiStreamRequest,
+  ClientStreamRequest,
+  IdempotentUnaryRequest,
+  ServerStreamRequest,
+  UnaryRequest,
+} from "../gen/connectrpc/conformance/v1/service_pb.js";
+
+export function createTransport(req: ClientCompatRequest) {
+  let scheme = "http://";
+  if (req.serverTlsCert.length > 0) {
+    scheme = "https://";
+  }
+  const baseUrl = `${scheme}${req.host}:${req.port}`;
+
+  switch (req.httpVersion) {
+    case HTTPVersion.HTTP_VERSION_1:
+    case HTTPVersion.HTTP_VERSION_2:
+      break;
+    case HTTPVersion.HTTP_VERSION_3:
+      throw new Error("HTTP/3 is not supported");
```

Comment:
Would definitely add a TODO here, since it raises a lot of questions, see https://github.com/connectrpc/connect-es/pull/1019#discussion_r1491178819